lua/starfall/libs_sv/prop.lua

```diff
@@ -430,9 +458,57 @@ function props_library.createSent(pos, ang, class, frozen)
 		end
 
 		hookcall = "PlayerSpawnedVehicle"
-
+	elseif sent2 then
+		if ply ~= NULL then
+			if (scripted_ents.GetStored(class).t.AdminOnly and not ply:IsAdmin()) then SF.Throw("This sent is admin only!", 2) end
+			if gamemode.Call("PlayerSpawnSENT", ply, class) == false then SF.Throw("Another hook prevented the sent from spawning", 2) end
+		end
+		
+		local enttbl = {}
+		data = data or {}
+		
+		-- Apply data
+		for param, org in pairs(sent2[1]) do
+			local value = data[param]
+			
+			if value then
+				value = ounwrap(value) or value
+				
+				checkluatype(value, org[2], nil, "Parameter: " .. param)
+				
+				enttbl[org[1]] = value
+			else
+				enttbl[org[1]] = org[3]
```

Comment:
Is there always a default? I think an error would be better in some cases so users know they are misusing the SENT.

Reply:
I've added ability for default to be missing. I'd recommend removing defaults that don't make sense so that the user must provide that data.